subprojects/packagefiles/oatpp/test/meson.build

```diff
@@ -0,0 +1,121 @@
+    'oatpp/core/provider/PoolTest.cpp',
+    'oatpp/core/provider/PoolTest.hpp',
+    'oatpp/encoding/Base64Test.cpp',
+    'oatpp/encoding/Base64Test.hpp',
+    'oatpp/encoding/UnicodeTest.cpp',
+    'oatpp/encoding/UnicodeTest.hpp',
+    'oatpp/network/ConnectionPoolTest.cpp',
+    'oatpp/network/ConnectionPoolTest.hpp',
+    'oatpp/network/UrlTest.cpp',
+    'oatpp/network/UrlTest.hpp',
+    'oatpp/network/monitor/ConnectionMonitorTest.cpp',
+    'oatpp/network/monitor/ConnectionMonitorTest.hpp',
+    'oatpp/network/virtual_/InterfaceTest.cpp',
+    'oatpp/network/virtual_/InterfaceTest.hpp',
+    'oatpp/network/virtual_/PipeTest.cpp',
+    'oatpp/network/virtual_/PipeTest.hpp',
+    'oatpp/parser/json/mapping/DTOMapperPerfTest.cpp',
+    'oatpp/parser/json/mapping/DTOMapperPerfTest.hpp',
+    'oatpp/parser/json/mapping/DTOMapperTest.cpp',
+    'oatpp/parser/json/mapping/DTOMapperTest.hpp',
+    'oatpp/parser/json/mapping/DeserializerTest.cpp',
+    'oatpp/parser/json/mapping/DeserializerTest.hpp',
+    'oatpp/parser/json/mapping/EnumTest.cpp',
+    'oatpp/parser/json/mapping/EnumTest.hpp',
+    'oatpp/parser/json/mapping/UnorderedSetTest.cpp',
+    'oatpp/parser/json/mapping/UnorderedSetTest.hpp',
+    'oatpp/web/ClientRetryTest.cpp',
+    'oatpp/web/ClientRetryTest.hpp',
+    'oatpp/web/FullAsyncClientTest.cpp',
+    'oatpp/web/FullAsyncClientTest.hpp',
+    'oatpp/web/FullAsyncTest.cpp',
+    'oatpp/web/FullAsyncTest.hpp',
+    'oatpp/web/FullTest.cpp',
+    'oatpp/web/FullTest.hpp',
+    'oatpp/web/PipelineAsyncTest.cpp',
+    'oatpp/web/PipelineAsyncTest.hpp',
+    'oatpp/web/PipelineTest.cpp',
+    'oatpp/web/PipelineTest.hpp',
+    'oatpp/web/app/BasicAuthorizationController.hpp',
+    'oatpp/web/app/BearerAuthorizationController.hpp',
+    'oatpp/web/app/Client.hpp',
+    'oatpp/web/app/Controller.hpp',
+    'oatpp/web/app/ControllerAsync.hpp',
+    'oatpp/web/app/ControllerWithInterceptors.hpp',
+    'oatpp/web/app/ControllerWithInterceptorsAsync.hpp',
+    'oatpp/web/app/DTOs.hpp',
+    'oatpp/web/mime/multipart/StatefulParserTest.cpp',
+    'oatpp/web/mime/multipart/StatefulParserTest.hpp',
+    'oatpp/web/protocol/http/encoding/ChunkedTest.cpp',
+    'oatpp/web/protocol/http/encoding/ChunkedTest.hpp',
+    'oatpp/web/server/HttpRouterTest.cpp',
+    'oatpp/web/server/HttpRouterTest.hpp',
+    'oatpp/web/server/ServerStopTest.cpp',
+    'oatpp/web/server/ServerStopTest.hpp',
+    'oatpp/web/server/api/ApiControllerTest.cpp',
+    'oatpp/web/server/api/ApiControllerTest.hpp',
+    'oatpp/web/server/handler/AuthorizationHandlerTest.cpp',
+    'oatpp/web/server/handler/AuthorizationHandlerTest.hpp',
+)
+
+
+all_tests = executable(
+    'oatpp_all_tests',
+    test_src_files,
+    cpp_args: flags,
+    dependencies: [oatpp_dep, oatpp_test_dep],
+)
+# Note that these are extensive tests and take a long time! It takes ~ 5 Min on fast machines
+test('oatpp all tests', all_tests, timeout: 600)
```

Comment:
I would say around 900 seconds, these are extensive tests and In Linux on my machine they run in under 300 seconds. but I have a powerful machine. In my VM for windows they took mostly around 500 secs, but sometimes, when many things happened in the background up to ~800 secs :sweat: 

I figured that meson has an option test-multiplier, but it's not passable as CLI arg, or have I misread something ther?

Reply:
It's possible with the PR I just merged that adds test_options to the ci_config.json file.

Reply:
Also I wonder how parallel the upstream tests are. Meson can automatically parallelize tests if they are run as separate units, but this test bundles everything in one binary because the main() function runs every single test one after the other. :D Did not check further to see if it has e.g. threading.

Reply:
The tests use threading extensively, they spawn many threads and run them in parallel, so single threading is likely to reduce the performance of the tests massively

Reply:
Gotcha, so nothing to solve there.

I guess it really is just doing "that much work".